source/Ship.cpp

```diff
@@ -702,8 +720,23 @@ void Ship::FinishLoading(bool isNewInstance)
 		}
 		else
 			++it;
-		if(bay.side == Bay::INSIDE && bay.launchEffects.empty() && Crew())
-			bay.launchEffects.emplace_back(GameData::Effects().Get("basic launch"));
+
+		if(bay.side == Bay::INSIDE) {
+			bool bayRemoved = false;
+
+			if(partsToRemove && (partsToRemove->find("bay") != partsToRemove->end())) {
+				auto baysToRemove = partsToRemove->at("bay");
+
+				if(baysToRemove > 0) {
+					it = bays.erase(it - 1);
+					baysToRemove--;
+					bayRemoved = true;
+				}
+			}
+
+			if (!bayRemoved && bay.launchEffects.empty() && Crew())
+				bay.launchEffects.emplace_back(GameData::Effects().Get("basic launch"));
+		}
```

Comment:
This should not be limited to removing only certain bays. It is already possible to redefine to a different bay configuration, as long as some bays are present, so removing some bays with this syntax is unnecessary, and also very limiting, since it doesn't allow removing any specific bays, just whichever ones are first in the collection.
This whole part shouldn't be present.